ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,95 @@
+package ch.cyberduck.core.ctera;
+
+import ch.cyberduck.core.Acl;
+import ch.cyberduck.core.DefaultIOExceptionMappingService;
+import ch.cyberduck.core.DisabledListProgressListener;
+import ch.cyberduck.core.Local;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.PathAttributes;
+import ch.cyberduck.core.dav.DAVExceptionMappingService;
+import ch.cyberduck.core.dav.DAVPathEncoder;
+import ch.cyberduck.core.dav.DAVSession;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.shared.DefaultAclFeature;
+
+import org.w3c.dom.Element;
+
+import javax.xml.namespace.QName;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import com.github.sardine.impl.SardineException;
+import com.github.sardine.util.SardineUtil;
+
+import static ch.cyberduck.core.ctera.CteraCustomACL.allCteraCustomACLRoles;
+import static ch.cyberduck.core.ctera.CteraCustomACL.toQn;
+
+
+public class CteraAclPermissionFeature extends DefaultAclFeature {
```

Comment:
Implement `AclPermission` instead of extending from abstract implementation if all methods are overriden.